cmd/services.rb

```diff
@@ -42,9 +43,46 @@ def services
 
     raise UsageError, "`brew services` is supported only on macOS!" unless OS.mac?
 
+    # pbpaste's exit status is a proxy for detecting the use of reattach-to-user-namespace
+    raise UsageError, "`brew services` cannot run under tmux!" if ENV["TMUX"] && !quiet_system("/usr/bin/pbpaste")
+
     # Keep this after the .parse to keep --help fast.
     require_relative "../lib/services_cli"
 
-    Homebrew::ServicesCli.run!(args)
+    # Parse arguments.
+    subcommand, formula, = args.named
+
+    if ["list", "cleanup"].include?(subcommand)
+      raise UsageError, "The `#{subcommand}` subcommand does not accept a formula argument!" if formula
+      raise UsageError, "The `#{subcommand}` subcommand does not accept the --all argument!" if args.all?
+    end
+
+    target = if args.all?
+      available_services
+    elsif formula
+      Service.new(Formulary.factory(formula))
+    end
+
+    # Dispatch commands and aliases.
+    case subcommand.presence
+    when nil, "list", "ls"
+      Homebrew::ServicesCli.list
```

Comment:
Could consider putting some of these subcommands in separate files?

Reply:
Yeah, this cleanup is heavily inspired by `brew bundle` so I was already thinking about it. I'll do that later though, I want to get the Linux support done first.